### PR TITLE
Duplicate donors search and merge

### LINKED ISSUE #350: Add API endpoints to find and merge duplicate donors
Add an API endpoint that will search the database and find all duplicate donors matching on last name, first name, date of birth and gender.

Create a wizard that allows the user to select values from the various duplicate Donor and merge them into one.

In the backend, the following happens after a merge is finished:
 * a new donor will be created
 * the merged donors will be given the status 'MERGED' which means they will not appear in any donor searches
 * the donations and deferrals from the merged donors will be moved to the new donor
 * a backup record of which donations and deferrals were moved is kept in the database table 'DuplicateDonorBackup'
 * the blood testing rules engine is run on the donations after they have been merged

### PR DESCRIPTION
**Pull request by timler**
Closes #350

### CODE REVIEW

**Comment by timler** [comment]
@danfuterman 

Please see the 1st attempt for a algorithm to find the duplicates. As discussed it is very simple but is contained in a separate service so it can be refactored easily.

I have a few questions regarding matching with one Donor has a null value for one of the attributes I am using to match (perhaps it's not even a possibility). My code is null safe, but if for example Donor#1 has no gender but Donor#2 is male (and all else matches) - is that a match or not?

**Comment by timler** [comment]
@danfuterman please could you review the new table I added and also perhaps the endpoints? I will resolve the merge conflicts next week.

**Comment by timler** [comment]
@danfuterman looking good I believe. It's ready to merge.

File: src/service/BloodTestsService.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	BloodTestingRepository bloodTestingRepository;
+	
+	/**
+	 * Executes the BloodTestingRuleEngine with the configured BloodTests and returns the results
+	 * 
+	 * @param donor Donor who made the donation
+	 * @param donation Donation to run the tests on
+	 * @return BloodTestingRuleResult with the results from the tests
+	 */
+	public BloodTestingRuleResult executeTests(Donor donor, Donation donation) {
```
Comment (bausmeier) [inline src/service/BloodTestsService.java]: @timler When I was looking to use this method in my code I noticed that the donor param isn't used.

**Comment by danfuterman** [comment]
@bausmeier - if/when you get some time, it would be great if you could also review this, along with the associated frontend PR https://github.com/jembi/bsis-frontend/pull/185.

**Comment by timler** [comment]
@danfuterman Added the new permissions as specified in the ticket. See  9275b11

**Comment by danfuterman** [comment]
@bausmeier Can you please do a final code review of this (if you haven't already) before I merge the PR in. Thanks. I am happy with everything as is.

**Comment by bausmeier** [comment]
@danfuterman Looks fine to me.

**Pull request merged**